src/rpp/rpp/operators/flat_map.hpp

```diff
@@ -0,0 +1,76 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2023 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/operators/map.hpp>
+#include <rpp/operators/merge.hpp>
+#include <rpp/defs.hpp>
+
+namespace rpp::operators::details
+{
+
+template<rpp::constraint::decayed_type Fn>
+struct flat_map_t 
+{
+    RPP_NO_UNIQUE_ADDRESS Fn m_fn;
+
+    template<rpp::constraint::observable TObservable>
+        requires (std::invocable<Fn, rpp::utils::extract_observable_type_t<TObservable>> 
+                    && rpp::constraint::observable<std::invoke_result_t<Fn, rpp::utils::extract_observable_type_t<TObservable>>>)
+    auto operator()(TObservable&& observable) const &
```

Comment:
Can you add test to cover this overloading? (it is enough to apply operator from variable in any test case)

Reply:
done